apps/prairielearn/src/pages/publicAssessmentQuestions/publicAssessmentQuestions.ts

```diff
@@ -16,19 +19,20 @@ router.get(
   asyncHandler(async (req, res) => {
     const assessment_id = req.params.assessment_id;
     const course_instance_id = req.params.course_instance_id;
-    const isAssessmentPublic = await checkAssessmentPublic(assessment_id);
+    const isAssessmentPublic = await selectAssessmentIsPublic(assessment_id);
     if (!isAssessmentPublic) {
       throw new error.HttpStatusError(404, 'Not Found');
     }
 
     const course = await selectCourseByCourseInstanceId(course_instance_id.toString());
     res.locals.course = course;
-    res.locals.assessment = await selectAssessmentById(assessment_id);
+    const assessment = await selectAssessmentById(assessment_id);
+    res.locals.assessment = assessment;
+
+    assert(assessment.assessment_set_id);
+    const assessment_set_name = (await selectAssessmentSetById(assessment.assessment_set_id)).name;
```

Comment:
Nit, conventionally we'd pass around full objects as long as we can. That is, we should avoid picking off `name` here and just pass the whole assessment set object down to the HTML template.